src/wiktextract/type_utils.py

```diff
@@ -0,0 +1,14 @@
+from typing import (
+    Union,
+)
+
+
+WordData = dict[str, Union[
```

Comment:
I think `list[dict[str, any]]` is fine for `parse_page()`, since these functions are not directly called and the data are written to JSON file soon after. Plus mypy doesn't work well with union types.

And could you please format the newly added code and rebase the commits instead of merge master branch?

Reply:
What's the difference between rebasing and merging here?

Reply:
rebase is like the branch is created at the current master branch so the commits will looks like they are created linearly otherwise old commits could appear anywhere after merge and it's harder to read the git log history, and rebase don't have distracting merge commits. Git book has some nice pictures: https://git-scm.com/book/en/v2/Git-Branching-Rebasing 

Basically I think it's like changing your start point to another branch, like the current master branch but not the outdated branch. And also has other features like squash commits and reword a commit message.

Reply:
Tried rebasing, take a look at the commit history now. :+1: I will soon undo this and revert to previous commits, but suffice to say I will not be using rebase for this branch. I might restart the branch just to keep things simple and try, *try* if rebasing makes any sense then.

Reply:
That's much better now, however there are still two extra merge commits. You could create a new branch from the `wiktextract-typing` branch locally and experimenting on that branch, then delete the old  `wiktextract-typing` and rename the new branch.

A comment to https://github.com/tatuylonen/wiktextract/pull/432/commits/0d41c02efb894a1d77265bbb71238e93f2016608: the long  `HEAD_TAG_RE` was created in a function before is because language data are loaded later from JSON files so they are not available from the beginning.

Reply:
I've reverted to before rebasing, the duplicate commits are now gone.

Reply:
I mean "were". Language data only come from the new package now.

My sentences are pron to have typos and grammatical errors if I'm not careful...

Reply:
Phew, I thought I might have missed something.

Reply:
I don't understand how could you merge these commits and start over?

Reply:
I've merged the branch and now I'll start a new branch from master. Then I can use rebase to keep the branch updated without weird duplicate commits that happened when I rebased previously when I already had merges there. When I rebased previously, there were suddenly 18 commits instead of the actual 12, with duplicates. I would probably have to undo all the merges somehow to make a 'clean' branch and rebase that.

Continuing making small type checking changes in a new branch should be fine. It's not like it's actually code that breaks things.

Reply:
I didn't think of the merge commits, not sure how to rebase in that case...

Reply:
You could start a new branch from the remote master branch and then cherry-pick individual commits from your working branch. 

It's a bit of work and a good lesson for the future to keep the git history clean. 😁 A lesson I had to learn myself recently. 😇